### PR TITLE
Cover more types for C_API

### LINKED ISSUE #1863: Export global vars into c_api
# Summary

Hello.

Faiss contains various external global variables.
here
https://github.com/facebookresearch/faiss/blob/master/faiss/utils/distances.h#L156-L167

and here
https://github.com/facebookresearch/faiss/blob/master/faiss/IndexIVF.h#L412

Is this API available to library users?

Running on:
- CPU

Interface: 
- C++

### PR DESCRIPTION
**Pull request by ava57r**
Exported some global variables and statistics.
Supported downcast for IndexIDMap and IndexIDMap2 from faiss::Index
Fixes #1863

### CODE REVIEW

**Comment by mdouze** [comment]
Sounds good. Maybe @Enet4 wants to take a look?

File: c_api/MetaIndexes_c.h [hunk 1]
```diff
@@ -25,6 +25,14 @@ FAISS_DECLARE_GETTER_SETTER(IndexIDMap, int, own_fields)
 
 int faiss_IndexIDMap_new(FaissIndexIDMap** p_index, FaissIndex* index);
 
+/** attempt a dynamic cast to a IDMap, thus checking
+ * check whether the underlying index type is `IndexIDMap`.
```
Comment (Enet4) [inline c_api/MetaIndexes_c.h:29]:
```suggestion
/** attempt a dynamic cast to a IDMap, thus checking
 * whether the underlying index type is `IndexIDMap`.
```

File: c_api/MetaIndexes_c.h [hunk 2]
```diff
@@ -40,13 +48,23 @@ void faiss_IndexIDMap_id_map(
...
+
+FAISS_DECLARE_GETTER_SETTER(IndexIDMap2, int, own_fields)
 
 int faiss_IndexIDMap2_new(FaissIndexIDMap2** p_index, FaissIndex* index);
 
 /// make the rev_map from scratch
 int faiss_IndexIDMap2_construct_rev_map(FaissIndexIDMap2* index);
 
+/** attempt a dynamic cast to a IDMap2, thus checking
+ * check whether the underlying index type is `IndexIDMap`.
```
Comment (Enet4) [inline c_api/MetaIndexes_c.h:61]:
```suggestion
 * whether the underlying index type is `IndexIDMap`.
```

**Review by Enet4 — COMMENTED** [review]
Looks OK to me. All I found was a few extraneous words in the documentation blocks.

**Pull request closed without merging**